src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts

```diff
@@ -628,7 +628,8 @@ class TerminalTabsDragAndDrop implements IListDragAndDrop<ITerminalInstance> {
 						sourceInstances = [instance];
 						this._terminalService.moveToTerminalView(instance);
 					} else if (workspaceId !== this._workspaceContextService.getWorkspace().id) {
-						await this._localTerminalService.requestDetachInstance(workspaceId, Number.parseInt(instanceId));
+						this._localTerminalService.requestDetachInstance(workspaceId, Number.parseInt(instanceId));
+						await timeout(300);
```

Comment:
The request ID and response idea should fix this more reliably

Reply:
Oh also we should await it, no?

Reply:
even with awaiting it, it won't work because it needs time to propagate from the terminal service to the ptyService

Reply:
the request id and response idea from what I understand will not prevent us from having to ask the terminal service for the correct persistentProcessId

Reply:
Return a promise like this from `requestDetachInstance` so that it's done after it's been awaited:

https://github.com/Microsoft/vscode/blob/b0cde4e27114c229b385c5fa3a2b9d6ed09d8d84/src/vs/platform/terminal/node/ptyHostService.ts#L313-L319